src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -496,8 +498,21 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter, ID
         await this.dumpCell(cell.document.uri.toString());
 
         if (this.configuration.__mode === KernelDebugMode.RunByLine) {
+            const textLines = cell.document.getText().split('\r\n');
```

Comment:
.splitLines({trim: false, removeEmptyEntries: false}) might be the safer option here.